packages/cadl-ranch-specs/http/server/version/union/main.tsp

```diff
@@ -0,0 +1,66 @@
+import "@typespec/http";
+import "@typespec/versioning";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-azure-core";
+
+using Azure.Core;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+
+@doc("Illustrates server with customer-defined and template-defined api versions.")
+@scenarioService("/server/version/union")
+@useDependency(Azure.Core.Versions.v1_0_Preview_1)
+namespace Server.Version.Union;
+
+@doc("Pet")
+@resource("Pet")
+model Pet {
+  @key
+  @visibility("read")
+  @doc("pet name")
+  name: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing purpose"
+@doc("Server with multiple api versions and operation with customer-defined api version.")
+@scenario
+@scenarioDoc("""
+Server with multiple api versions and operation with customer-defined api version.
+
+Expected path with api version parameter: /server/version/union/customerDefined?api-version=1.0.0
+""")
+@route("/customerDefined")
+@head
+op customerDefined(
+  @doc("Api version.")
+  @query
+  apiVersion: string
+): NoContentResponse;
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing purpose"
+@doc("Server with multiple api versions and operation with customer-defined api version.")
+@scenario
+@scenarioDoc("""
+Server with multiple api versions and operation with customer-defined api version.
+
+Expected path with api version parameter: /server/version/union/customerDefined?api-version=2.0.0
+""")
+@route("/customerDefinedWithDefault")
+@head
+op customerDefinedWithDefault(
+  @doc("Api version.")
+  @query
+  apiVersion: "2.0.0"
```

Comment:
Question: is any other language plan to support this?

Reply:
Why not?

Reply:
Why should service allowed to specify api-version that potentially different to the global one?

Reply:
I agree that all the api version should be defined globally.

Reply:
See the definition of Operation:
```
op Operation<
  TParams extends TypeSpec.Reflection.Model,
  TResponse,
  Traits extends TypeSpec.Reflection.Model = {},
  TErrorResponse = Azure.Core.Foundations.ErrorResponse
>(
  ...TraitProperties<
    Traits & VersionParameterTrait<ApiVersionParameter>,
    TraitLocation.ApiVersionParameter
  >,
  ...TParams
): TResponse | TErrorResponse;
```

Reply:
I mean the API version value should be defined as a global enum.